### PR TITLE
Fix/webmodeler resize

### PR DESCRIPTION
**Pull request by djkwagala**

### CODE REVIEW

File: src/TimeSeries.webmodeler.ts [hunk 1]
```diff
@@ -12,12 +12,24 @@ export class preview extends Component<TimeSeriesContainerProps, {}> {
...
             createElement(Alert, { message })
         );
     }
 
+    componentDidMount() {
+        this.setUpEvents();
+    }
+
+    componentWillUnmount() {
+        const modelerIFrame = this.getIframe();
```
Comment (acellam) [inline src/TimeSeries.webmodeler.ts:27]: Add space after declaring variables

File: src/TimeSeries.webmodeler.ts [hunk 2]
```diff
@@ -116,12 +128,25 @@ export class preview extends Component<TimeSeriesContainerProps, {}> {
         }
         return errorMessage;
     }
+
+    private setUpEvents() {
+        const modelerIFrame = this.getIframe();
```
Comment (acellam) [inline src/TimeSeries.webmodeler.ts:133]: Add space after block of variable declaration

File: src/components/Alert.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
-import { DOM, StatelessComponent } from "react";
+import { StatelessComponent, createElement } from "react";
 
 export const Alert: StatelessComponent<{ message?: string }> = (props) =>
     props.message
-        ? DOM.div({ className: "alert alert-danger widget-time-series-alert" }, props.message)
+        ? createElement("div", { className: "alert alert-danger widget-time-series-alert" }, props.message)
         : null as any;
```
Comment (acellam) [inline src/components/Alert.ts]: remove `as any`

File: src/components/Alert.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import { DOM, StatelessComponent } from "react";
+import { StatelessComponent, createElement } from "react";
 
 export const Alert: StatelessComponent<{ message?: string }> = (props) =>
     props.message
-        ? DOM.div({ className: "alert alert-danger widget-time-series-alert" }, props.message)
```
Comment (acellam) [inline src/components/Alert.ts]: Make component generic by passing the  css class to allow it to be used for other alert types ie danger, success,

File: src/components/NVD3LineChart.ts [hunk 5]
```diff
@@ -33,7 +33,7 @@ class NVD3LineChart extends Component<Nvd3LineChartProps, {}> {
     static defaultProps: Partial<Nvd3LineChartProps> = { datum: [] };
     private chart: LineChart;
     private resizeHandler: Nvd3ResizeHandler;
-    private svg: SVGElement;
+    private svg: Element;
```
Comment (acellam) [inline src/components/NVD3LineChart.ts]: `svg` is supposed to remain `SVGElement` element

File: src/components/NVD3LineChart.ts [hunk 6]
```diff
@@ -49,8 +49,8 @@ class NVD3LineChart extends Component<Nvd3LineChartProps, {}> {
             style.height = `${this.props.height}%`;
         }
 
-        return DOM.div({ className: "widget-time-series nv-chart", style },
-            DOM.svg({ ref: node => this.svg = node })
+        return createElement("div", { className: "widget-time-series nv-chart", style },
+            createElement("svg", { ref: node => this.svg = node })
```
Comment (acellam) [inline src/components/NVD3LineChart.ts]: when `svg` above is SVGElement, u can make the node be returned as SVGElement by doing this
`ref: node => this.svg = node as SVGElement`

**Review by acellam — CHANGES_REQUESTED** [review]

File: src/components/NVD3LineChart.ts (hunk 5 above)
Comment (djkwagala) [reply src/components/NVD3LineChart.ts]: Done

File: src/components/NVD3LineChart.ts (hunk 6 above)
Comment (djkwagala) [reply src/components/NVD3LineChart.ts]: Done

File: src/TimeSeries.webmodeler.ts (hunk 2 above)
Comment (djkwagala) [reply src/TimeSeries.webmodeler.ts:133]: not necessary

File: src/TimeSeries.webmodeler.ts (hunk 1 above)
Comment (djkwagala) [reply src/TimeSeries.webmodeler.ts:27]: The space is not necessary in this case.

File: src/components/Alert.ts (hunk 3 above)
Comment (djkwagala) [reply src/components/Alert.ts]: done

**Pull request closed without merging**